src/sentry/static/sentry/app/components/projects/missingProjectMembership.tsx

```diff
@@ -47,101 +53,176 @@ class MissingProjectMembership extends React.Component<Props, State> {
+        <StyledButton
+          priority="primary"
+          type="button"
+          onClick={this.joinTeam.bind(this, team)}
+        >
           {t('Join Team')}
-        </a>
+        </StyledButton>
       );
     }
     return (
-      <a className="btn btn-default" onClick={this.joinTeam.bind(this, team)}>
+      <StyledButton
+        priority="primary"
+        type="button"
+        onClick={this.joinTeam.bind(this, team)}
+      >
         {t('Request Access')}
-      </a>
+      </StyledButton>
     );
   }
 
-  renderExplanation(features: Set<string>) {
-    if (features.has('open-membership')) {
-      return t('To view this data you must one of the following teams.');
-    } else {
-      return t(
-        'To view this data you must first request access to one of the following teams:'
-      );
-    }
-  }
-
-  renderJoinTeams(features: Set<string>) {
+  getTeamsForAccess() {
+    const request: string[] = [];
+    const pending: string[] = [];
     const teams = this.state.project?.teams ?? [];
+    const team = teams.map(tm => TeamStore.getBySlug(tm.slug));
+    team.map(tm => (tm?.isPending ? pending.push(tm!.slug) : request.push(tm!.slug)));
+    return [request, pending];
+  }
 
-    if (!teams.length) {
-      return (
-        <EmptyMessage>
-          {t(
-            'No teams have access to this project yet. Ask an admin to add your team to this project.'
-          )}
-        </EmptyMessage>
-      );
-    }
+  handleChange = (evt: string) => {
+    const input = evt;
+    this.setState({team: input});
+  };
 
-    return teams.map((team: Team) => (
-      <p key={team.slug}>
-        #{team.slug}: {this.renderJoinTeam(team, features)}
-      </p>
-    ));
-  }
+  renderPendingTeam = (team: string) => {
+    return {
+      value: team,
+      label: <DisabledLabel>{`#${team}`}</DisabledLabel>,
+    };
+  };
 
   render() {
-    const {organization} = this.props;
+    const {organization, groupId} = this.props;
+    const team = this.state.team;
+    const teams = this.state.project?.teams ?? [];
     const features = new Set(organization.features);
 
+    const teamAccess = [
+      {
+        label: t('Request Access'),
+        options: this.getTeamsForAccess()[0].map(request => ({
+          value: request,
+          label: `#${request}`,
+        })),
+      },
+      {
+        label: t('Pending Requests'),
+        options: this.getTeamsForAccess()[1].map(pending =>
+          this.renderPendingTeam(pending)
+        ),
+      },
+    ];
+
     return (
       <div className="container">
-        <StyledWell centered>
-          <StyledIconFlag size="xxl" />
-          <p>{t("You're not a member of this project.")}</p>
-          <p>{this.renderExplanation(features)}</p>
-          {this.renderJoinTeams(features)}
-        </StyledWell>
+        <Panel>
+          {!teams.length ? (
+            <EmptyMessage icon={<StyledIconFlag size="xl" color="gray300" />}>
```

Comment:
I'm looking at `<EmptyMessage>` and it looks like it provides a default icon color and size ([the `font-size` is controlled by a prop to `EmptyMessage`](https://github.com/getsentry/sentry/blob/7d36ffa98147b0380dea1a7a203487384ea3d5ac/src/sentry/static/sentry/app/views/settings/components/emptyMessage.tsx#L55-L56))

In this case, I would leave `IconFlag` un-styled so that it is consistent across all uses of `<EmptyMessage>`. I could see an exception being made with the size depending on how it looks.

Reply:
Cool, I removed the styles but needed to keep the size since it was really small without it.